packages/@aws-cdk/aws-apigatewayv2-authorizers/lib/http/lambda.ts

```diff
@@ -0,0 +1,114 @@
+ */
+export enum HttpLambdaAuthorizerType {
+  /** Returns simple boolean response */
+  SIMPLE,
+
+  /** Returns an IAM Policy */
+  AWS_IAM,
+}
+
+/**
+ * Properties to initialize HttpTokenAuthorizer.
+ */
+export interface HttpLambdaAuthorizerProps {
+
+  /**
+   * The name of the authorizer
+   * @default 'LambdaAuthorizer'
+   */
+  readonly authorizerName?: string;
+
+  /**
+   * The identity source for which authorization is requested.
+   *
+   * @default ['$request.header.Authorization']
+   */
+  readonly identitySource?: string[];
+
+  /**
+   * Specifies the format of the payload sent to an HTTP API Lambda authorizer.
+   *
+   * @default - 2.0
+   */
+  readonly payloadFormatVersion?: AuthorizerPayloadFormatVersion;
+
+  /**
+   * The lambda function used for authorization
+   */
+  readonly handler: IFunction;
+
+  /**
+   * How long APIGateway should cache the results. Max 1 hour.
+   * Disable caching by setting this to 0.
+   *
+   * @default Duration.minutes(5)
+   */
+  readonly resultsCacheTtl?: Duration;
+
+  /**
+   * Type of lambda authorizer
+   */
+  readonly type: HttpLambdaAuthorizerType;
+}
+
+/**
+ * Authorize Http Api routes via a lambda function
+ */
+export class HttpLambdaAuthorizer implements IHttpRouteAuthorizer {
+  private authorizer?: HttpAuthorizer;
+
+  constructor(private readonly props: HttpLambdaAuthorizerProps) {
+    if (props.type === HttpLambdaAuthorizerType.SIMPLE && props.payloadFormatVersion === AuthorizerPayloadFormatVersion.VERSION_1_0) {
+      throw new Error('The simple authorizer type can only be used with payloadFormatVersion 2.0');
+    }
+  }
+
+  public bind(options: HttpRouteAuthorizerBindOptions): HttpRouteAuthorizerConfig {
+    if (!this.authorizer) {
+      const id = this.props.authorizerName && !Token.isUnresolved(this.props.authorizerName) ?
+        this.props.authorizerName : 'LambdaAuthorizer';
+
+      this.authorizer = new HttpAuthorizer(options.scope, id, {
+        httpApi: options.route.httpApi,
+        identitySource: this.props.identitySource ?? [
+          '$request.header.Authorization',
+        ],
+        type: HttpAuthorizerType.LAMBDA,
+        authorizerName: this.props.authorizerName,
+        enableSimpleResponses:
+          this.props.type === HttpLambdaAuthorizerType.SIMPLE,
+        payloadFormatVersion:
+          this.props.payloadFormatVersion ??
+          AuthorizerPayloadFormatVersion.VERSION_2_0,
+        authorizerUri: lambdaAuthorizerArn(this.props.handler),
+        resultsCacheTtl: this.props.resultsCacheTtl ?? Duration.minutes(5),
+      });
+    }
+
+    return {
+      authorizerId: this.authorizer.authorizerId,
+      authorizationType: HttpRouteAuthorizerType.LAMBDA,
+    };
+  }
+}
+
+/**
+ * constructs the authorizerURIArn.
+ */
+function lambdaAuthorizerArn(handler: IFunction) {
+  return `arn:${Stack.of(handler).partition}:apigateway:${Stack.of(handler).region}:lambda:path/2015-03-31/functions/${handler.functionArn}/invocations`;
```

Comment:
Use `Stack.formatArn()`.

Reply:
Seems like that's not possible for lambda, `Stack.formatArn()` injects account id in the wrong place breaking the format.

This is the format required via [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigatewayv2-authorizer.html#cfn-apigatewayv2-authorizer-authorizeruri):

```
arn:aws:apigateway:us-east-1:lambda:path/2015-03-31/functions/arn:aws:lambda:us-east-1:{account_id}:function:{lambda_function_name}/invocations
```

Using `formatArn` with:

```ts
function lambdaAuthorizerArn(handler: IFunction) {
  return Stack.of(handler).formatArn({
    service: 'apigateway',
    resource: `lambda:path/2015-03-31/functions/${handler.functionArn}/invocations`,
  });
}

// arn:aws:apigateway:us-east-1:<account-id>:lambda:path/2015-03-31/functions/arn:aws:lambda:us-east-1:<account-id>:function:AuthorizerInteg-authfunction96361832-WFURCN8B7G4U/invocations
```

As you can see it injects account id right after region. That doesn't exist in the required format. [Assuming this is why apigw v1 authorizer does this](https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-apigateway/lib/authorizers/lambda.ts#L257)

Reply:
Got it. It's fine then. Even though the string starts with `arn:`, looks like it's not actually an ARN.